overlord/configstate/configcore/picfg_test.go

```diff
@@ -166,6 +168,94 @@ func (s *piCfgSuite) TestConfigurePiConfigRegression(c *C) {
 	s.checkMockConfig(c, expected)
 }
 
+func (s *piCfgSuite) TestUpdateConfigUC20RunMode(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	defer func() { dirs.SetRootDir("") }()
+
+	// mock the device as uc20 run mode
+	mockCmdline := filepath.Join(dirs.GlobalRootDir, "cmdline")
+	err := ioutil.WriteFile(mockCmdline, []byte("snapd_recovery_mode=run"), 0644)
+	c.Assert(err, IsNil)
+	restore = osutil.MockProcCmdline(mockCmdline)
+	defer restore()
+
+	// write default config at both the uc18 style runtime location and uc20 run
+	// mode location to show that we only modify the uc20 one
+	piCfg := filepath.Join(boot.InitramfsUbuntuSeedDir, "config.txt")
+	uc18PiCfg := filepath.Join(dirs.GlobalRootDir, "/boot/uboot/config.txt")
+
+	err = os.MkdirAll(filepath.Dir(piCfg), 0755)
+	c.Assert(err, IsNil)
+	err = os.MkdirAll(filepath.Dir(uc18PiCfg), 0755)
+	c.Assert(err, IsNil)
+
+	err = ioutil.WriteFile(piCfg, []byte(mockConfigTxt), 0644)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(uc18PiCfg, []byte(mockConfigTxt), 0644)
+	c.Assert(err, IsNil)
+
+	// apply the config
+	err = configcore.Run(&mockConf{
+		state: s.state,
+		conf: map[string]interface{}{
+			"pi-config.gpu-mem-512": true,
+		},
+	})
+	c.Assert(err, IsNil)
+
+	// make sure that the original pi config.txt in /boot/uboot/config.txt
+	// didn't change
+	c.Check(uc18PiCfg, testutil.FileEquals, mockConfigTxt)
+
+	// but the real one did change*
+	expected := strings.Replace(mockConfigTxt, "#gpu_mem_512=true", "gpu_mem_512=true", -1)
+	c.Check(piCfg, testutil.FileEquals, expected)
+}
+
+func (s *piCfgSuite) testUpdateConfigUC20NonRunMode(c *C, mode string) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	defer func() { dirs.SetRootDir("") }()
```

Comment:
(nitpick) this is not needed, piCfgSuite is using configcoresuite and that sets dirs.SetRootDir() and adds the cleanup too.

Reply:
done